#!/usr/bin/env python

PACKAGE = 'mixmcl'

from math import pi
from dynamic_reconfigure.parameter_generator_catkin import ParameterGenerator, int_t, double_t, str_t, bool_t

gen = ParameterGenerator()

########################MCMCL########################
gen.add("dual_normalizer_ita"  , double_t, 0, "MCMCL parameters", 0.0001, 0.0, 1.0) 
gen.add("demc_factor_gamma", double_t, 0, "MCMCL parameters", 0.95, 0.0, 2.0) 
gen.add("demc_loc_bandwidth"   , double_t, 0, "MCMCL parameters", 0.01, 0.0, 1.0) 
gen.add("demc_ori_bandwidth"   , double_t, 0, "MCMCL parameters", 0.1, 0.0, 1.0) 
gen.add("position_bandwidth"   , double_t, 0, "MixMCL parameters", 10.0, 0.0, 100.0) 
gen.add("oriantation_bandwidth", double_t, 0, "MixMCL parameters", 0.4, 0.0, 10.0) 
######################end MCMCL######################

exit(gen.generate(PACKAGE, "mcmcl_node", "MCMCL"))
